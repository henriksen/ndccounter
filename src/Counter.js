import React, { useState } from 'react';
import styles from './Counter.module.css';

function Counter() {
  const [count, setCount] = useState(0);

  const canDecrement = () => {
    return count > 0;
  };

  return (
    <div className={styles.counter}>
      <button onClick={() => setCount(count - 1)} disabled={!canDecrement()}>
        -
      </button>
      <span aria-label="counter">{count}</span>
      <button onClick={() => setCount(count + 1)}>+</button>
    </div>
  );
}

export default Counter;
