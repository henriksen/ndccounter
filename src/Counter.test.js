import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-testing-library';
import Counter from './Counter';
const config = { baseUrl: 'https://localhost:5001' };
beforeEach(cleanup);

it('renders correctly', () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should not be able to decrement initially', () => {
  const { getByText } = render(<Counter />);
  expect(getByText('-').attributes.getNamedItem('disabled')).toBeDefined();
});

it('should be able to increment initially', () => {
  const { getByText } = render(<Counter config={config} />);
  expect(getByText('+').attributes.getNamedItem('disabled')).toBeNull();
});

it('should increment the number when clicking the + button', () => {
  const { getByText, getByLabelText } = render(<Counter config={config} />);
  fireEvent.click(getByText('+'));
  expect(getByLabelText('counter').innerHTML).toBe('1');
});

it('should be able to decrement after incrementing', () => {
  const { getByText } = render(<Counter config={config} />);
  fireEvent.click(getByText('+'));
  expect(getByText('-').attributes.getNamedItem('disabled')).toBeNull();
});

it('should decrement the number when clicking the - button', () => {
  const { getByText, getByLabelText } = render(<Counter config={config} />);
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('-'));
  expect(getByLabelText('counter').innerHTML).toBe('1');
});
