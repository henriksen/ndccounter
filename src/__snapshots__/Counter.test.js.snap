// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="counter"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    -
  </button>
  <span
    aria-label="counter"
  >
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
